src/js/tech/flash.js

```diff
@@ -485,8 +485,16 @@ Flash.checkReady = function(tech) {
 // Trigger events from the swf on the player
 Flash.onEvent = function(swfID, eventName) {
   const tech = Dom.getEl(swfID).tech;
-
-  tech.trigger(eventName, Array.prototype.slice.call(arguments, 2));
+  const args = Array.prototype.slice.call(arguments, 2);
+
+  // dispatch Flash events after asynchronously for two reasons:
+  // - Flash swallows any exceptions generated by javascript it
+  //   invokes
+  // - Flash is suspended until the javascript returns which may cause
+  //   playback performance issues
+  setTimeout(function() {
```

Comment:
Maybe use `tech.setTimeout()`? It's theoretically possible that the tech would be disposed in between `onEvent` getting called at the timeout running.